Error in websocket connection from client causes app to crash

**What steps did you take and what happened:**
2020-09-11T18:07:41.923Z	ERROR	api/websocket_service.go:43	create websocket client	{"err": "websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header"}
github.com/vmware-tanzu/octant/internal/api.serveWebsocket
	/__w/octant/octant/internal/api/websocket_service.go:43
github.com/vmware-tanzu/octant/internal/api.websocketService.func1
	/__w/octant/octant/internal/api/websocket_service.go:35
net/http.HandlerFunc.ServeHTTP
	/usr/local/go/src/net/http/server.go:2012
github.com/vmware-tanzu/octant/internal/api.rebindHandler.func1.1
	/__w/octant/octant/internal/api/middleware.go:121
net/http.HandlerFunc.ServeHTTP
	/usr/local/go/src/net/http/server.go:2012
github.com/gorilla/mux.(*Router).ServeHTTP
	/go/pkg/mod/github.com/gorilla/mux@v1.7.4/mux.go:210
github.com/vmware-tanzu/octant/pkg/octant.noCacheRootMiddleware.func1
	/__w/octant/octant/pkg/octant/options.go:85
net/http.HandlerFunc.ServeHTTP
	/usr/local/go/src/net/http/server.go:2012
github.com/gorilla/mux.(*Router).ServeHTTP
	/go/pkg/mod/github.com/gorilla/mux@v1.7.4/mux.go:210
github.com/gorilla/handlers.(*cors).ServeHTTP
	/go/pkg/mod/github.com/gorilla/handlers@v1.4.2/cors.go:138
net/http.serverHandler.ServeHTTP
	/usr/local/go/src/net/http/server.go:2807
net/http.(*conn).serve
	/usr/local/go/src/net/http/server.go:1895
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x17a1c62]

goroutine 248 [running]:
github.com/vmware-tanzu/octant/internal/api.(*WebsocketClient).readPump.func1(0x0)
	/__w/octant/octant/internal/api/websocket_client.go:128 +0x22
panic(0x1a9c8a0, 0x45f3360)
	/usr/local/go/src/runtime/panic.go:967 +0x15d
github.com/vmware-tanzu/octant/internal/api.(*WebsocketClient).readPump(0x0)
	/__w/octant/octant/internal/api/websocket_client.go:140 +0x7f
created by github.com/vmware-tanzu/octant/internal/api.serveWebsocket
	/__w/octant/octant/internal/api/websocket_service.go:47 +0xac


**What did you expect to happen:**
While making errored ws connection, octant crashes, and cause app to go down, panic needed to be handled without causing app crash.

**Anything else you would like to add:**
Add panic recover to server.


**Environment:**

Version:  0.15.0
Git commit:  97a507c2b071764933bb479007822672d5fa19f5
Built:  2020-08-12T16:45:29Z

--- a/changelogs/unreleased/1340-wwitzel3
+++ b/changelogs/unreleased/1340-wwitzel3
@@ -1,0 +1,1 @@
+Fix panic when WebsocketClientManager.ClientFromRequest returns an error

--- a/internal/api/websocket_service.go
+++ b/internal/api/websocket_service.go
@@ -39,9 +39,11 @@ func websocketService(manager ClientManager, dashConfig config.Dash) http.Handle
 func serveWebsocket(manager ClientManager, dashConfig config.Dash, w http.ResponseWriter, r *http.Request) {
 	client, err := manager.ClientFromRequest(dashConfig, w, r)
 	if err != nil {
-		logger := dashConfig.Logger()
-		logger.WithErr(err).Errorf("create websocket client")
-
+		if dashConfig != nil {
+			logger := dashConfig.Logger()
+			logger.WithErr(err).Errorf("create websocket client")
+		}
+		return
 	}
 
 	go client.readPump()

--- a/internal/api/websocket_service_test.go
+++ b/internal/api/websocket_service_test.go
@@ -1,0 +1,22 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/vmware-tanzu/octant/internal/config"
+)
+
+type fakeWebsocketClientManager struct {
+	WebsocketClientManager
+}
+
+func (c *fakeWebsocketClientManager) ClientFromRequest(dashConfig config.Dash, w http.ResponseWriter, r *http.Request) (*WebsocketClient, error) {
+	return nil, fmt.Errorf("test: error")
+}
+
+func TestWebsocketService_serveWebsocket(t *testing.T) {
+	f := &fakeWebsocketClientManager{}
+	serveWebsocket(f, nil, nil, nil)
+}

